src/sshfs_mount/sftp_server.cpp

```diff
@@ -334,6 +338,22 @@ int mp::SftpServer::mapped_gid_for(const int gid)
     return gid;
 }
 
+int mp::SftpServer::reverse_uid_for(const int uid, const int rev_uid_if_not_found)
+{
+    auto found = std::find_if(uid_map_list.cbegin(), uid_map_list.cend(),
+                              [uid](std::pair<int, int> p) { return uid == p.second; });
+
+    return found == uid_map_list.cend() ? rev_uid_if_not_found : found->first;
+}
+
+int mp::SftpServer::reverse_gid_for(const int gid, const int rev_gid_if_not_found)
+{
+    auto found = std::find_if(gid_map_list.cbegin(), gid_map_list.cend(),
+                              [gid](std::pair<int, int> p) { return gid == p.second; });
+
+    return found == gid_map_list.cend() ? rev_gid_if_not_found : found->first;
+}
```

Comment:
This code is pretty redundant and seems like it can be refactored into an anonymous function that accepts the map, id, and default id.  Something like:
```c++
namespace
{
int reverse_id_for(const id_mappings& id_maps, const int id, const int rev_id_if_not_found)
{
    auto found = std::find_if(id_maps.cbegin(), id_maps.cend(), [&id](std::pair<int, int> p) {
        return id == p.second;
    });

    return found == id_maps.cend() ? rev_id_if_not_found : found->first;
}
}
```

Reply:
Right, done. Refactored forward and reverse mapping functions.